### ISSUE TITLE
UI issues and Algorithms mess

### ISSUE THREAD

**Issue by zim96** [boeyjw/dsksch#6]
UI's size is too big giving issues to those running it under 900p screens and scaling results in problems in information not being displayed.

UI design is not desirable, it is tough to input 50+ requests. Disk scheduling performance only matters when there are large number of requests to resemble real world scenarios.

Class layout is a mess, it would prove to be a nightmare to maintain this code. Having 2 algorithms in 1 class isn't good design.

**Comment by boeyjw** [comment]
I agree with the UI design. It was not tested thoroughly with all generic resolutions and thus, this is the result.

I don't understand what you meant by "tough to input 50+ request" and how it even matters with the performance with the program. Are you speaking in the context of UI design or the program performance?

As for class design, the classes would look dumb if there was only 1 algorithm per class as it would be ridiculously short and makes no sense. Doing so just increases the complexity of maintenance. In this assignment, every algorithm has a precedence and its upgraded version. Thus it makes sense to pair up algorithm which are similar.

**Comment by zim96** [comment]
There should be an easy way to input custom requests, having hard locked presets isn't very useful for software that is supposed to simulate scenarios. For instance, 50+ requests should be a good benchmark for heavy loads. Your design to input custom requests is just a chore to use.

According to observations made, simulation softwares usually read from file. Perhaps that route should have been taken up instead?

It would have made more sense to have classes inherit properties of similar ones. For example, CSCAN, LOOK inherits SCAN. CLOOK inherits LOOK. This follows the principle of refactoring, classes should be small.

**Comment by boeyjw** [comment]
What you are doing is way out of the assignment scope. I do know your idealistic nature to achieve it at best you can but we simply do not that much time and resources to go beyond what we are assigned to do. We simply stick to the requirements.

Your design is feasible as well. I don't see how mine would be any more complicated. You prefer to do it in chronological order whereas I did it in a modular method.

**Comment by zim96** [comment]
The requirements was vague and I actually didn't spend more than 5 hours coding my implementation. Have you considered the maintenance that would come after development? Lets say someone wants to add a new algorithm or wants to compare the simulation of multiple files?
